task.ps1

```diff
@@ -28,15 +28,53 @@ New-AzSshKey -Name $sshKeyName -ResourceGroupName $resourceGroupName -PublicKey
 
 New-AzPublicIpAddress -Name $publicIpAddressName -ResourceGroupName $resourceGroupName -Location $location -Sku Basic -AllocationMethod Dynamic -DomainNameLabel $dnsLabel
 
-New-AzVm `
--ResourceGroupName $resourceGroupName `
--Name $vmName `
--Location $location `
--image $vmImage `
--size $vmSize `
--SubnetName $subnetName `
--VirtualNetworkName $virtualNetworkName `
--SecurityGroupName $networkSecurityGroupName `
--SshKeyName $sshKeyName  -PublicIpAddressName $publicIpAddressName
-
-# ↓↓↓ Write your code here ↓↓↓
+Write-Host "Creating network interface..."
+$nic = New-AzNetworkInterface -ResourceGroupName $resourceGroupName -Location $location -Name "${vmName}Nic" `
+    -SubnetId (Get-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName).Subnets[0].Id `
+    -PublicIpAddressId (Get-AzPublicIpAddress -Name $publicIpAddressName -ResourceGroupName $resourceGroupName).Id `
+    -NetworkSecurityGroupId (Get-AzNetworkSecurityGroup -Name $networkSecurityGroupName -ResourceGroupName $resourceGroupName).Id
+
+Write-Host "Creating VM $vmName ..."
+$cred = New-Object System.Management.Automation.PSCredential ("azureuser", (ConvertTo-SecureString "placeholderpassword" -AsPlainText -Force))
+$vmConfig = New-AzVmConfig -VMName $vmName -VMSize $vmSize | `
+    Set-AzVMOperatingSystem -Linux -ComputerName $vmName -Credential $cred -DisablePasswordAuthentication | `
+    Set-AzVMSourceImage -PublisherName "Canonical" -Offer "0001-com-ubuntu-server-jammy" -Skus "22_04-lts-gen2" -Version "latest" | `
+    Add-AzVMNetworkInterface -Id $nic.Id | `
+    Add-AzVMSSHPublicKey -KeyData $sshKeyPublicKey -Path "/home/azureuser/.ssh/authorized_keys"
+
+Write-Host "Starting VM creation..."
+$vm = New-AzVm -ResourceGroupName $resourceGroupName -Location $location -VM $vmConfig
```

Comment:
New VM should be created with `-SshKeyName` flag. This SshKey should be created earlier in the script.